src/freenet/client/filter/M3UFilter.java

```diff
@@ -141,7 +141,7 @@ public void readFilter(InputStream input, OutputStream output, String charset, H
                                 // impaired.
                                 if (cb instanceof GenericReadFilterCallback) {
                                     try {
-                                        filtered = ((GenericReadFilterCallback)cb).makeURIAbsolute(filtered);
+                                        filtered = ((GenericReadFilterCallback)cb).makeURIAbsoluteWithHostAndPort(filtered);
```

Comment:
Oh, and there is no test for this, either. So before adding new tests for this feature you would have to fix the test which relies on an incorrect base URI being handed in to the filter in order to test the URL resolving.

Reply:
This should now be fixed; I now use the host (IP+Port) from the request headers.